src/construction.cpp

```diff
@@ -1297,6 +1297,18 @@ static vpart_id vpart_from_item( const itype_id &item_id )
     return vpart_frame_vertical_2;
 }
 
+static vpart_id vpart_appliance_from_item( const itype_id &item_id )
+{
+    for( const auto &e : vpart_info::all() ) {
```

Comment:
```suggestion
    for( const std::pair<vpart_id, vpart_info> &e : vpart_info::all() ) {
```
Try to avoid auto for types that are not very complicated